File: src/dotnet/Ansight.Tools.Database/DatabaseOptionsBuilderExtensions.cs
namespace Ansight.Tools.Database;

using System;

public static class DatabaseOptionsBuilderExtensions
{
    public static Options.OptionsBuilder WithDatabaseTools(this Options.OptionsBuilder builder)
    {
        ArgumentNullException.ThrowIfNull(builder);

        return builder.AddTools(new ITool[]
        {
            new ListDatabasesTool(),
            new DescribeSchemaTool(),
            new QueryDatabaseTool()
        });
    }
}

File: src/android/ansight-tools-database/src/main/kotlin/ai/ansight/tools/database/DatabaseOptionsBuilderExtensions.kt
package ai.ansight.tools.database

import ai.ansight.runtime.AnsightOptionsBuilder

fun AnsightOptionsBuilder.withDatabaseTools(): AnsightOptionsBuilder {
    return addTools(AndroidDatabaseTools.create())
}

fun AnsightOptionsBuilder.withDatabaseTools(
    options: AndroidDatabaseToolsOptions,
): AnsightOptionsBuilder {
    return addTools(AndroidDatabaseTools.create(options))
}

fun AnsightOptionsBuilder.withDatabaseTools(
    configure: AndroidDatabaseToolsOptionsBuilder.() -> Unit,
): AnsightOptionsBuilder {
    val builder = AndroidDatabaseToolsOptions.createBuilder()
    builder.configure()
    return withDatabaseTools(builder.build())
}
